<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Promise</title>
</head>

<body>
  <h5>promise is an object that will produce one value some time in the feture promise</h5>
  <script>
    //  two type of declear promise: 1(".then"), 2("async")

    function getOrder() {
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          const chease = "🧀";
          resolve(chease);
          // reject("error occured")
        }, 2000);
      });
    };
    function getChese(chease) {
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          const doshe = chease + "🥞";
          resolve(doshe);
        }, 2000);
      })
    }
    function makePizza(doshe) {
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          const pizza = "🍕";
          resolve(pizza);
        }, 2000);
      })
    }

    async function getRadyPizza() {
      try {
        const cheas = await getOrder();
        console.log("here it is ", cheas);
        const doshe = await getChese(cheas);
        console.log("here it is ready ", doshe);
        const pizza = await makePizza();
        console.log("ready pizza ", pizza);
      }
      catch (error) {
        console.error("error occured");
      }

    }
    getRadyPizza();
    // fetch () api call two ways 

    fetch('https://jsonplaceholder.typicode.com/users/10').then((respos) => {
      return respos.json();
    }).then((data) => {
      console.log(data);
    }).catch(() => {
      console.log("error");
    })





    // async function you can get fetch api data
    // async function getApi() {
    //   try {
    //     const response = await fetch('https://jsonplaceholder.typicode.com/users/10');
    //     const data = await response.json();
    //     console.log(data);
    //     console.log(data.username);
    //   } catch (error) {
    //     console.log("error");
    //   }


    // }
    // getApi();





    // then method()
    // getOrder().then((chease) => {
    //   console.log("chease is here", chease);
    //   return getChese(chease);
    // }).then((doshe) => {
    //   console.log("doshe is here ", doshe);
    //   return makePizza(doshe);
    // }).then((pizza) => {
    //   console.log("pizza id ready", pizza);
    // }).catch((error) => {
    //   console.log("erroe occured");
    // })















  </script>
</body>

</html>







<!-- // // new Promise((resolve, reject) => {
// //   setTimeout(() => {
// //     const num = Math.floor(Math.random() * 5);
// //     console.log(num);
// //     if (num <= 3) {
// //       resolve();
// //     } else {
// //       reject("we want to error");
// //     }
// //   }, 2000);
// // })
// //   .then(() => {
// //     console.log("is done");
// //   })
// //   .catch((eror) => {
// //     console.log(eror);
// //   });

// // fetch:-
// const response = fetch("https://randomuser.me/api/")
// //   .then((response) => {
// //     console.log(response);
// //   })
// //   .catch(() => {
// //     console.log("eror");
// //   });
// //

// const promiseThree = new Promise((resolve, reject) => {
//   setTimeout(() => {
//     let erro = true;
//     if (!erro) {
//       resolve({
//         userName: "Aditya sarkar",
//         password: "1213",
//       });
//     } else {
//       reject("ERROR :js went wrong");
//     }
//   }, 1000);
// });
// promiseThree
//   .then((user) => {
//     return user;
//   })
//   .then((item) => {
//     console.log(item);
//   })
//   .catch(() => {
//     console.log("how is josh");
//   }); -->